### PR TITLE
fix method of FDR threshholding in GLM

### PR DESCRIPTION
**Pull request by ajthr**
Fix GH-2879

### CODE REVIEW

File: nilearn/glm/thresholding.py [hunk 1]
```diff
@@ -98,8 +98,7 @@ def fdr_threshold(z_vals, alpha):
     z_vals_ = - np.sort(- z_vals)
     p_vals = norm.sf(z_vals_)
     n_samples = len(p_vals)
-    pos = p_vals < alpha * np.linspace(
-        .5 / n_samples, 1 - .5 / n_samples, n_samples)
+    pos = p_vals < alpha * np.linspace(1/n_samples, 1,n_samples)
```
Comment (bthirion) [inline nilearn/glm/thresholding.py:101]:
```suggestion
    pos = p_vals < alpha * np.linspace(1 / n_samples, 1, n_samples)
``` We need to enforce pep8 conventions !

**Review by bthirion — COMMENTED** [review]
Thx ! 
Could you add a unit test to enforce this correction ? This is important to avoid a regression in the future. Thx !

**Comment by AjithRamachandran** [comment]
> Thx !
> Could you add a unit test to enforce this correction ? This is important to avoid a regression in the future. Thx !

Sure :)

File: nilearn/glm/thresholding.py [hunk 2]
```diff
@@ -98,8 +98,8 @@ def fdr_threshold(z_vals, alpha):
     z_vals_ = - np.sort(- z_vals)
     p_vals = norm.sf(z_vals_)
     n_samples = len(p_vals)
-    pos = p_vals < alpha * np.linspace(
-        .5 / n_samples, 1 - .5 / n_samples, n_samples)
+    pos = p_vals < alpha * np.linspace(1 / n_samples, 1, n_samples)
+``` We need to enforce pep8 conventions !
```
Comment (NicolasGensollen) [inline nilearn/glm/thresholding.py:102]: I think you added this line unintentionally. It is not properly indented and causes the test errors you're getting.

**Review by NicolasGensollen — COMMENTED** [review]
Thanks for opening this @AjithRamachandran !

File: nilearn/glm/thresholding.py (hunk 2 above)
Comment (AjithRamachandran) [reply nilearn/glm/thresholding.py:102]: Yes. I didnt see that there.

File: nilearn/glm/tests/test_thresholding.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 def test_fdr():
     rng = np.random.RandomState(42)
     n = 100
-    x = np.linspace(.5 / n, 1. - .5 / n, n)
+    x = np.linspace(1 / n, 1, n)
```
Comment (bthirion) [inline nilearn/glm/tests/test_thresholding.py:17]: Thx, but the idea is to have a test such that it fails with the previous version, and passes after the fix.

**Comment by bthirion** [comment]
I can try and figure out something.

**Comment by bthirion** [comment]
the cranch has been deleted. I'm going to redo the PR from scratch...

**Comment by bthirion** [comment]
superseded by #3137, closing

**Pull request closed without merging**